src/utilities/__tests__/resolveSchemaCoordinate-test.ts

```diff
@@ -0,0 +1,189 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+
+import type {
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+} from '../../type/definition.js';
+import type { GraphQLDirective } from '../../type/directives.js';
+
+import { buildSchema } from '../buildASTSchema.js';
+import { resolveSchemaCoordinate } from '../resolveSchemaCoordinate.js';
+
+describe('resolveSchemaCoordinate', () => {
+  const schema = buildSchema(`
+    type Query {
+      searchBusiness(criteria: SearchCriteria!): [Business]
+    }
+
+    input SearchCriteria {
+      name: String
+      filter: SearchFilter
+    }
+
+    enum SearchFilter {
+      OPEN_NOW
+      DELIVERS_TAKEOUT
+      VEGETARIAN_MENU
+    }
+
+    type Business {
+      id: ID
+      name: String
+      email: String @private(scope: "loggedIn")
+    }
+
+    directive @private(scope: String!) on FIELD_DEFINITION
+  `);
+
+  it('resolves a Named Type', () => {
+    expect(resolveSchemaCoordinate(schema, 'Business')).to.deep.equal({
+      kind: 'NamedType',
+      type: schema.getType('Business'),
+    });
+
+    expect(resolveSchemaCoordinate(schema, 'String')).to.deep.equal({
+      kind: 'NamedType',
+      type: schema.getType('String'),
+    });
+
+    expect(resolveSchemaCoordinate(schema, 'private')).to.deep.equal(undefined);
```

Comment:
should this be `null`?

```suggestion
    expect(resolveSchemaCoordinate(schema, 'private')).to.deep.equal(null);
```

Reply:
per @benjie's comment here https://github.com/magicmark/graphql-js/pull/1#discussion_r2122309237 I believe `undefined` is correct, despite the spec saying "null". I think a note _somewhere_ in the spec text saying:

> "This spec documents functions as yielding "null". In the reference graphql-js implementation, this corresponds to a value of `undefined`"

Reply:
Right, it'd help a lot 👍 Also, could we include the reason why we're doing this? Is there any advantage of using `undefined` instead of just `null`?

Reply:
`undefined` is what @leebyron's original implementation used. I don't have a strong opinion either way here, there's a valid semantic argument for either. I would be ok to switch. Do you feel strongly that it should be `null`?

Reply:
From an outside vantage point, it sure looks surprising that the spec says `null` but `graphql-js` returns `undefined`.

But also I'm not the biggest `graphql-js` connoisseur out there so I'll leave y'all to decide.

Reply:
@benjie wdyt?

Reply:
```
> [1, 2, 3].find(n => n === 4)
undefined
```

If it's a big deal, don't say `Return {null}` just say `Return the blah, if it exists.` and leave the "if it doesn't exist" up to the implementation. Which for the vast majority of implementations will be `null` but in JS it's more common to use `undefined` for "not found".

Reply:
The more I think about it the more this feels like the right answer. We shouldn't mandate `null` e.g. in Elixir you might want to return something like `:not_found` on failure vs `{:field, ...}` on success

Reply:
Addressed https://github.com/graphql/graphql-spec/pull/794/commits/5ac68a005b07b70a581574774cbd58b1971dd78a

Reply:
I'll leave it out for now to avoid more things to bikeshed, but I think it would be worth a note in the spec explaning this context above.

Prior art, we give implemenation notes [here](https://github.com/graphql/graphql-spec/blob/a44d4ec24e4c1263ab984a201cf9207c5a4dd549/spec/Section%203%20--%20Type%20System.md?plain=1#L2162-L2163)